sklearn/feature_extraction/_hashing_fast.pyx

```diff
@@ -27,17 +29,12 @@ def transform(raw_X, Py_ssize_t n_features, dtype,
     """
     assert n_features > 0
```

Comment:
If we are at it, could we make this cleaner and securer?

Reply:
I think this is a quick developer check. `FeatureHasher` already validates n_features here:

https://github.com/scikit-learn/scikit-learn/blob/231dd7b03b105e1c177dddfc8e4c2b5532cf2fb6/sklearn/feature_extraction/_hash.py#L112-L117

Reply:
I learned that one should not rely on `assert` outside of tests as they can be circumvented and that raising errors is to be preferred.

Reply:
n_features is guaranted to be positive here so this assert is useless anyway. Let's remove it